File: Prac6/s20426586.pl
#!/usr/bin/perl

use strict;
use warnings;

# Prompt the user for input file
print "Enter the input file name: ";
my $input_file = <STDIN>;
chomp $input_file;

# Check if the input file exists
unless (-e $input_file) {
    die "Error: File '$input_file' not found. Please try again.\n";
}

# Prompt the user for the word to search
print "Enter the word to search: ";
my $word_to_search = <STDIN>;
chomp $word_to_search;

# Open file or exit with error
open(my $fh, '<', $input_file) or die "Error: Cannot open file '$input_file': $!\n";

# Read file contents
my $file_contents = do { local $/; <$fh> };

# Convert to lowercase and strip non-alphabetic characters
$file_contents =~ tr/A-Za-z//cd;
$file_contents = lc $file_contents;

# Count occurrences of the word in reverse order
my $matches = () = reverse($file_contents) =~ /\Q$word_to_search\E/g;

# Output the result
print "Matches: $matches\n";

File: Prac6/s20426586.php
<?php
// Check if correct number of arguments are provided
if ($argc !== 3) {
    exit("Usage: php php_script.php <input_file> <word_to_search>\n");
}

// Read input file contents
$input_file = $argv[1];
$word_to_search = $argv[2];

// Check if file exists and is readable
if (!file_exists($input_file) || !is_readable($input_file)) {
    exit("error: I hate to be that guy, but the file is called 'input.txt', try again bro.\n");
}

// Read file contents
$file_contents = file_get_contents($input_file);

// Convert to lowercase and strip non-alphabetic characters
$file_contents = preg_replace("/[^a-zA-Z]/", "", strtolower($file_contents));

// Count occurrences of the word in reverse order
$matches = substr_count(strrev($file_contents), strrev($word_to_search));

// Output the result as HTML
echo "<!DOCTYPE html>\n";
echo "<html>\n";
echo "<head><title>PHP Script Output</title></head>\n";
echo "<body>\n";
echo "<h1>Matches: $matches</h1>\n";
echo "</body>\n";
echo "</html>\n";
?>
